### ISSUE TITLE
Why `train_controlnet_webdataset.py` uses `EulerDiscreteScheduler`?

### ISSUE THREAD

**Issue by okotaku** [huggingface/diffusers#4873]
I noticed that the train_controlnet_webdataset.py script in our codebase uses EulerDiscreteScheduler for training, whereas most other training scripts use DDPMScheduler. Furthermore, I observed that there are some differences between this script and the original train_controlnet_sdxl.py script.

I'm interested in understanding the rationale behind choosing EulerDiscreteScheduler for training in this particular script. Could someone please provide insights into why this scheduler was chosen and whether there are specific advantages or use cases where it outperforms DDPMScheduler? Additionally, clarification on the differences between train_controlnet_webdataset.py and the original train_controlnet_sdxl.py would be greatly appreciated.

Thank you for your assistance in clarifying this matter.

## schedulers difference

https://github.com/huggingface/diffusers/blob/7b93c2a882d8e12209fbaeffa51ee2b599ab5349/examples/controlnet/train_controlnet_sdxl.py#L811

https://github.com/huggingface/diffusers/blob/7b93c2a882d8e12209fbaeffa51ee2b599ab5349/examples/research_projects/controlnet/train_controlnet_webdataset.py#L960

## other difference (about sigmas)

https://github.com/huggingface/diffusers/blob/7b93c2a882d8e12209fbaeffa51ee2b599ab5349/examples/research_projects/controlnet/train_controlnet_webdataset.py#L1341-L1342
https://github.com/huggingface/diffusers/blob/7b93c2a882d8e12209fbaeffa51ee2b599ab5349/examples/research_projects/controlnet/train_controlnet_webdataset.py#L1368-L1369

Related PR: https://github.com/huggingface/diffusers/pull/4633
@sayakpaul @patil-suraj

**Comment by patrickvonplaten** [comment]
cc @williamberman as well here

**Comment by sayakpaul** [comment]
I would say @patil-suraj here.

**Comment by williamberman** [comment]
The most general way to look at it which class is used is just that you need a noise schedule that determines some standard deviation of normal distribution to sample from when adding noise to the latent image. At no point in any training scripts that we have do we use anything from the scheduler class other than adding noise. We do not ever use the step function to solve an {O,S}DE backwards or sample a posterior.

The only difference for why using the DDPMScheduler vs the EulerScheduler is that the EulerScheduler already has the code to give us the noise schedule in terms of the standard deviations which can be used for additional conditioning of the network and weighting of the loss function at different noise levels. I think it's like a one line change if we wanted to add the ability to get the sigmas out of the ddpm scheduler but as they're derived from the alphas and betas that it uses to parameterize its noise schedule.  

SDXL is trained to output a unit variance scaled noise. A very general form of a loss function used for diffusion models in from the EDM (karras) paper is where the predicted noise is appropriately scaled and combined with the noised image to come to a prediction for the unnoised image which is then used with a weighted mse for the loss. To do the most general form of this loss function (in its edm parameterization), you need direct access to the sigmas which is why we've been using it in some of our webdatasets training scripts. 

_Technically_ as the network is trained to just predict the unit variance scaled noise, you _can_ just directly train on the mse with the unit variance added noise. In that case, you don't need access to the sigmas. This is why some of the training scripts for sdxl can just use the DDPMScheduler.

**Comment by haoqiangyu** [comment]
> The most general way to look at it which class is used is just that you need a noise schedule that determines some standard deviation of normal distribution to sample from when adding noise to the latent image. At no point in any training scripts that we have do we use anything from the scheduler class other than adding noise. We do not ever use the step function to solve an {O,S}DE backwards or sample a posterior.
> 
> The only difference for why using the DDPMScheduler vs the EulerScheduler is that the EulerScheduler already has the code to give us the noise schedule in terms of the standard deviations which can be used for additional conditioning of the network and weighting of the loss function at different noise levels. I think it's like a one line change if we wanted to add the ability to get the sigmas out of the ddpm scheduler but as they're derived from the alphas and betas that it uses to parameterize its noise schedule.
> 
> SDXL is trained to output a unit variance scaled noise. A very general form of a loss function used for diffusion models comes from the EDM (karras) paper is where the predicted noise is appropriately scaled and combined with the noised image to come to a prediction for the unnoised image which is then used with a weighted mse for the loss. To do the most general form of this loss function (in its edm parameterization), you need direct access to the sigmas which is why we've been using it in some of our webdatasets training scripts.
> 
> _Technically_ as the network is trained to just predict the unit variance scaled noise, you _can_ just directly train on the mse with the unit variance added noise. In that case, you don't need access to the sigmas. This is why some of the training scripts for sdxl can just use the DDPMScheduler.

@williamberman Your answer is very professional, but I don't understand it at the moment. I want to ask a question, if the same model is trained using these two different schedulers, what will be the difference in the training results? Thanks!
